auditlog/models/rule.py

```diff
@@ -423,8 +429,14 @@ def write_fast(self, vals, **kwargs):
             old_vals2 = dict.fromkeys(list(vals2.keys()), False)
             old_values = {id_: old_vals2 for id_ in self.ids}
             new_values = {id_: vals2 for id_ in self.ids}
+            # Check if any value has actually changed
+            changes = any(
+                old_values[rec_id] != new_values[rec_id]
+                for rec_id in self.ids
+                if rec_id in old_values and rec_id in new_values
+            )
             result = write_fast.origin(self, vals, **kwargs)
-            if self.env.user in users_to_exclude:
+            if not changes or self.env.user in users_to_exclude:
```

Comment:
Please check if this doesn't skip log creation when a value is changed to false from something else.